K8SPXC-1407 Clarify HA Proxy upgrade risks (#153)
percona/k8spxc-docs: docs/update.md

[Custom Resource](operator.md) configuration file is set to `SmartUpdate`
(this is the default value and the recommended way for upgrades).

As an alternative, the `updateStrategy` key can be set to `RollingUpdate` and
`OnDelete`. You can find out more about it in the [appropriate section](update.md#more-on-upgrade-strategies).

!!! warning

    The upgrade covers various components of the cluster including PMM Client
    and HA Proxy, which may need additional attention.
    

>     *  It is [highly recommended](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html)

Reviewer: ```suggestion
    *  It is **highly recommended**
```

>         to upgrade PMM Server **before** upgrading PMM Client.

Reviewer: ```suggestion
        to [upgrade PMM Server](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html) **before** upgrading PMM Client.
```

>     * If you are using [custom configuration for HA Proxy](haproxy-conf.md#passing-custom-configuration-options-to-haproxy), 

Reviewer: ```suggestion
    * If you are using [custom configuration for HAProxy](haproxy-conf.md#passing-custom-configuration-options-to-haproxy), 
```

>         **before upgrading**, check the HA Proxy configuration file provided by

Reviewer: ```suggestion
        **before upgrading**, check the HAProxy configuration file provided by
```

    !!! warning

        The above command upgrades various components of the cluster including PMM Client and HA Proxy, which may need additional attention.

>         * If you are using [custom configuration for HA Proxy](haproxy-conf.md#passing-custom-configuration-options-to-haproxy), check [HA Proxy release information](https://www.haproxy.org/) to make sure that the version you are applying is compatible with your existing custom config.

Reviewer 2: @fiowro, the issue is that we occasionally add new options to the HAProxy configuration. If users are using a custom configuration for HAProxy, they need to check the default configuration at https://github.com/percona/percona-docker/blob/pxc-operator-1.14.0/haproxy/dockerdir/etc/haproxy/haproxy-global.cfg and incorporate any new options into their custom config.

For example, in this commit: https://github.com/percona/percona-docker/commit/f2b9f63e8c288d219522ea76ae6b113e2717a59c, we added two new options in different places. If these options are not included in the custom config, users may experience downtime.

The instructions are as follows:
1. check new conf before the update
2. add new options while the update into CR

    !!! warning

>         The above command upgrades various components of the cluster including PMM Client. If you didn't follow the [official recommendation](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html) to upgrade PMM Server before upgrading PMM Client, you can avoid PMM Client upgrade by removeing it from the list of images as follows:

Reviewer: ```suggestion
        The above command upgrades various components of the cluster including PMM Client. If you didn't follow the [official recommendation](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html) to upgrade PMM Server before upgrading PMM Client, you can avoid PMM Client upgrade by removing it from the list of images as follows:
```
